### PR TITLE
Add support for ffigen upgrades

### LINKED ISSUE #1429: [Bug] Should allow ffigen 10.x
### Describe the bug

frb's ffigen version validation requires ffigen <=9. ffigen 10.0.0 has been released https://pub.dev/packages/ffigen/changelog#1000 and should be supported

### Codegen logs with `RUST_LOG=debug` environment variable

```shell
N/A
```

### To Reproduce

_No response_

### Expected behavior

_No response_

### Generated binding code

_No response_

### OS

_No response_

### Version of `flutter_rust_bridge_codegen`

_No response_

### Flutter info

_No response_

### Version of `clang++`

_No response_

### Version of `ffigen`

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by aran**
## Changes

Fixes #1429

This softens the `ffigen` requirement. ensure_tool_available still ensures ffigen is available, but doesn't cap the version.

ffigen 11 introduces a breaking change (https://pub.dev/packages/ffigen/changelog#1100) which can be disabled with a new `ignore-source-errors` flag. Will see from CI if that flag is needed for frb. I saw unrelated errors from ./frb_internal on master.

## Checklist

- [X] An issue to be fixed by this PR is listed above.
- [ ] New tests are added to ensure new features are working. Please refer to [this page](https://cjycode.com/flutter_rust_bridge/guides/contributing/tip) to see how to add a test.
- [ ] `./frb_internal precommit --mode slow` (or `fast`) is run (it internal runs code generator, does auto formatting, etc).
- [ ] If this PR adds/changes features, documentations (in the `./website` folder) are updated.
- [ ] CI is passing. Please refer to [this page](https://cjycode.com/flutter_rust_bridge/guides/contributing/tip) to see how to solve a failed CI.

### CODE REVIEW

File: frb_codegen/assets/integration_template/rust_builder/pubspec.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ dependencies:
 
 dev_dependencies:
   ffi: ^2.0.2
-  ffigen: ^9.0.0
+  ffigen: ">=9.0.0"
```
Comment (fzyzcjy) [inline frb_codegen/assets/integration_template/rust_builder/pubspec.yaml:17]: nit: maybe we should `< 12.0.0` to be safe

File: frb_codegen/src/library/commands/ensure_tools_available.rs [hunk 2]
```diff
@@ -8,7 +8,7 @@ use std::str::FromStr;
 
 lazy_static! {
     pub(crate) static ref FFIGEN_REQUIREMENT: VersionReq =
-        VersionReq::parse(">= 8.0.0, < 10.0.0").unwrap();
+        VersionReq::parse(">= 8.0.0").unwrap();
```
Comment (fzyzcjy) [inline frb_codegen/src/library/commands/ensure_tools_available.rs:11]: nit: `< 12.0.0`

File: frb_example/dart_build_rs/pubspec.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@ description: flutter rust bridge example
 version: 1.0.0
 publish_to: none
 environment:
-  sdk: '>=3.2.0 <4.0.0'
+  sdk: '>=3.3.0 <4.0.0'
```
Comment (fzyzcjy) [inline frb_example/dart_build_rs/pubspec.yaml:6]: btw you may need to change ci.yaml's flutter/dart version in order to make ci happy

File: frb_codegen/src/library/commands/ensure_tools_available.rs (hunk 2 above)
Comment (aran) [reply frb_codegen/src/library/commands/ensure_tools_available.rs:11]: What do you think about leaving this open? In the rarer case that a new `ffigen` version has an issue, user can always still fix it in their own `pubspec.yaml`. Or, the flutter_rust_bridge package itself in frb_dart can enforce this constraint. Most of the time new ffigen doesn't seem to cause a problem, so it's annoying to wait and synchronize with a new FRB version to allow the new ffigen when simply wanting to run `flutter pub upgrade --major-versions`

Comment (fzyzcjy) [reply frb_codegen/src/library/commands/ensure_tools_available.rs:11]: Looks reasonable!

Comment (aran) [reply frb_codegen/src/library/commands/ensure_tools_available.rs:11]: Sounds good, I will leave it open here, and place the constraint in frb_dart/pubspec.yaml.

File: frb_example/dart_build_rs/pubspec.yaml (hunk 3 above)
Comment (aran) [reply frb_example/dart_build_rs/pubspec.yaml:6]: Thanks for the tip, done.

**Review by fzyzcjy — APPROVED** [review]

**Pull request merged**